lib/nvme/nvme_internal.h

```diff
@@ -307,6 +307,12 @@ struct nvme_tracker {
 	LIST_ENTRY(nvme_tracker)	list;
 
 	struct nvme_request		*req;
+#if INTPTR_MAX == INT32_MAX
+	int32_t __pad[3];
```

Comment:
This adds 12B. Why not just add 4 only ?
I would also prefer that you do the padding using uint8_t and spell out __padding.

Reply:
Because we need to get to exactly 4096 bytes, and this is the sufficient number needed to do it?

I can spell things out I guess.

Reply:
OK. But at least add a comment and change the __padding field to be uint8_t.
I still think that these alignments are not needed and the asserts should go away. But that is a bigger change.

Reply:
The other reason to use int32_t is that it is the same as intptr_t in this case. But yeah, I guess it's the same on [12].